### PR TITLE
refactor(MenuItem): refactor MenuItem component to accommodate icons and custom components

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**MenuItem**
Github project ticket: #350

### CODE REVIEW

File: src/core/DropdownMenu/index.tsx [hunk 1]
```diff
@@ -8,31 +8,48 @@ import {
...
+  sdsIconProps?: Partial<IconProps<keyof IconNameToSmallSizes>>;
 }
 
+export interface DropdownMenuOptionComponent extends DropdownMenuOptionGeneral {
+  component?: ReactNode;
+}
+
+type Exclusive<T, U> = T & { [K in Exclude<keyof U, keyof T>]?: undefined };
+
+export type DefaultDropdownMenuOption =
```
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:48]: nice TypeScript magic 🔥 !

File: src/core/DropdownMenu/index.tsx [hunk 2]
```diff
@@ -220,29 +237,34 @@ const DropdownMenu = <
...
           )}
-        </StyledMenuItem>
-      </li>
+        </StyledMenuItemText>
+      );
+    }
+
+    return (
+      <MenuItem
+        column={option.count}
```
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:259]: nit: should we destruct `option`?

e.g., 

```ts
const { count, sdsIcon, ... } = option;

<MenuItem 
  column={count}
  ...
/>
```

File: src/core/InputDropdown/style.ts [hunk 3]
```diff
@@ -244,7 +244,13 @@ const isDisabled = (props: InputDropdownProps): SerializedStyles => {
   `;
 };
 
-const doNotForwardProps = ["intent", "open", "sdsStage", "isMinimal"];
+const doNotForwardProps = [
+  "intent",
+  "open",
+  "sdsStage",
+  "isMinimal",
+  "shouldTruncateMinimalDetails",
```
Comment (tihuan) [inline src/core/InputDropdown/style.ts:252]: ah sorry for missing this one 😆

File: src/core/InputDropdown/style.ts [hunk 4]
```diff
@@ -328,7 +334,9 @@ interface MinimalDetailsProps extends CommonThemeProps {
   shouldTruncateMinimalDetails: InputDropdownProps["shouldTruncateMinimalDetails"];
 }
 
-export const MinimalDetails = styled("div")`
+export const MinimalDetails = styled("div", {
+  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
```
Comment (tihuan) [inline src/core/InputDropdown/style.ts:338]: thank you for adding this 🙏

File: src/core/MenuItem/index.tsx [hunk 5]
```diff
@@ -1,38 +1,89 @@
...
   ContentWrapper,
   StyledCheck,
   StyledMenuItem,
+  StyledMenuItemIcon,
   TextWrapper,
 } from "./style";
 
-export interface MenuItemExtraProps {
+/* 
+  The final goal is to create a typescript interface which only accepts the
```
Comment (tihuan) [inline src/core/MenuItem/index.tsx:14]: great comment! Thank you!

File: src/core/MenuItem/index.tsx [hunk 6]
```diff
@@ -1,38 +1,89 @@
...
+
+/* 
+  In the second step we create a TypeScript interface that is a subset of 
+  the IconNameToSizes interface, containing only those icons that can be "xs" or "s" in size.
+
+  As a result, the MenuItem component accepts a bacteria icon with a 
+  size value of "xs" | "s" | "l" | "xl" or a check icon with a size value of "xs" | "s", 
+  but not a book icon since it has a size value of "l" | "xl".
+ */
+export type IconNameToSmallSizes = {
```
Comment (tihuan) [inline src/core/MenuItem/index.tsx:40]: v nice!

File: src/core/MenuItem/index.tsx [hunk 7]
```diff
@@ -41,8 +92,19 @@ const MenuItem = forwardRef((props: MenuItemProps, _) => {
           className="primary-text"
           disabled={disabled}
         >
+          {sdsIcon && (
+            <StyledMenuItemIcon disabled={disabled}>
+              <Icon
+                {...sdsIconProps}
+                sdsType="static"
+                sdsIcon={sdsIcon}
+                sdsSize="s"
```
Comment (tihuan) [inline src/core/MenuItem/index.tsx:101]: just wanted to double check this is intended to be hardcoded as `s` and not accepting `xs` as an option

**Review by tihuan — APPROVED** [review]
Thanks for adding the type checking, Masoud 🙏 Appreciate you fixing some leftover bugs from our InputDropdown PR too 😆 LGTM with some minor comments!

File: src/core/DropdownMenu/index.tsx (hunk 2 above)
Comment (masoudmanson) [reply src/core/DropdownMenu/index.tsx:259]: Yes, absolutely. I missed this one. 🫠

File: src/core/InputDropdown/style.ts (hunk 4 above)
Comment (masoudmanson) [reply src/core/InputDropdown/style.ts:338]: 🥳 🎉

File: src/core/MenuItem/index.tsx (hunk 5 above)
Comment (masoudmanson) [reply src/core/MenuItem/index.tsx:14]: This one got a little complicated 😵‍💫

File: src/core/MenuItem/index.tsx (hunk 7 above)
Comment (masoudmanson) [reply src/core/MenuItem/index.tsx:101]: MenuItems could only display an icon of size "s", according to the design file. I'll confirm with @clarsen-czi.

**Review by clarsen-czi — APPROVED** [review]
LGTM! Thanks @masoudmanson!!

File: src/core/MenuItem/index.tsx (hunk 7 above)
Comment (masoudmanson) [reply src/core/MenuItem/index.tsx:101]: As confirmed by @clarsen-czi, MenuItems only accept IconS.

**Pull request merged**